### PR TITLE
Feature/softmax activation

### PR DESCRIPTION
**Pull request by VIVelev**
The multi-class (vector) [Softmax activation function](https://en.wikipedia.org/wiki/Softmax_function) was implemented. Again, as `nujo built-in` function.

I tried the computation of the Jacobian matrix (the backward pass) to be as efficient as possible. It supports batch gradient descent. Open for suggestions on how to improve efficiency.

(Ignore docs updates)

### CODE REVIEW

**Review by VIVelev — COMMENTED** [review]
LGTM

**Review by ManiacMaxo — APPROVED** [review]
